test/middleware/company.authentication.middleware.spec.ts

```diff
@@ -0,0 +1,69 @@
+jest.mock("ioredis");
+jest.mock("@companieshouse/web-security-node");
+
+import { describe, expect, test, jest, beforeEach } from '@jest/globals';
+import { Request, Response, NextFunction } from 'express';
+import {
+  getSessionRequestWithExtraData,
+  COMPANY_NUMBER
+} from '../__mocks__/session.mock';
+import { companyAuthentication } from "../../src/middleware/company.authentication.middleware";
+import { authMiddleware } from "@companieshouse/web-security-node";
+import { logger } from '../../src/utils/logger';
+import { ANY_MESSAGE_ERROR } from '../__mocks__/text.mock';
+
+jest.mock('../../src/utils/logger', () => {
+  return {
+    logger: { debugRequest: jest.fn(), infoRequest: jest.fn(), errorRequest: jest.fn() }
+  };
+});
+
+const req = {} as Request;
+const res = { locals: {}, redirect: jest.fn() as any } as Response;
+const next = jest.fn();
+const mockCompanyAuthMiddleware = authMiddleware as jest.Mock;
+mockCompanyAuthMiddleware.mockImplementation(() => {
+  return (req: Request, res: Response, next: NextFunction) => {
+    return next();
+  };
+});
+
+describe('Company Authentication middleware', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    res.locals = {};
+  });
+
```

Comment:
Do we need tests on some of the update pages with no auth code previously entered?

Reply:
@AndreKramer7654  with navigation checks we won't be able to test this, and it will need an IT if we need to check if the page is accessed without auth code...

Reply:
You could mock out the navigation check (to pass) but maybe this is not needed.

Reply:
have mocked the navigation check as this as its now added to all the pages